scripts/smc/verify.js

```diff
@@ -0,0 +1,121 @@
+const exit = require('process').exit;
+const minimist = require('minimist');
+
+const verify = (siteId, name, secretKey) => {
+    return new Promise(async (resolve, _) => {
+        const body = { name };
+        const response = await fetch(`https://smc-config-api.kube.searchspring.io/api/customer/${siteId}/verify`, {
+            method: 'POST',
+            body: JSON.stringify(body),
+            headers: {
+                'accept': 'application/json',
+                'User-Agent': '',
+                'Authorization': `${secretKey}`
+            }
+        });
+        const data = await response.json();
+        if(data.message === 'success') {
+            console.log(`Authentication successful for siteId ${siteId}`)
+            resolve(true)
+        } else {
+            console.log(`Authentication failed for siteId ${siteId}`)
+            resolve(false)
+        }
+    });
+}
+
+(async function () {
+    try {
+        const argv = minimist(process.argv.slice(2),  { '--': true });
+
+        if(argv['siteId_Type'] !== 'string' && argv['siteId_Type'] !== 'object') {
+            console.log("Verify script requires 'siteId_Type' parameter with values either 'string' (single site) or 'object' (multi site)")
+            exit(1);
+        }
+        if(!argv['siteId_Type'] === 'string' && (!argv['siteId'] || !argv['repository'] || !argv['secretKey'])) {
+            console.log("Verify script requires 'siteId', 'repository' and 'secretKey' parameter")
+            exit(1);
+        }
+        if(!argv['siteId_Type'] === 'object' && (!argv['siteIds'] || !argv['siteNames'] || !argv['secrets-ci'])) {
+            console.log("Verify script requires 'siteId', 'repository' and 'secrets-ci' parameter")
+            exit(1);
+        }
+
+        let authFailed = false;
+        if(argv.siteId_Type === 'string') {
+            // single site
+            const siteId = argv.siteId;
+            const name = argv.repository;
+            const secretKey = argv.secretKey;
+
+            const success = await verify(siteId, name, secretKey);
+            if(!success) {
+                authFailed = true;
+            }
+        } else if (argv.siteId_Type === 'object') {
+            // multi site
+            const siteIds = argv.siteIds.split(',').filter(a => a);
+            const siteNames = argv.siteNames.split(',').filter(a => a);
+
+            if(siteIds.length !== siteNames.length) {
+                console.log("The amount of siteIds and siteNames does not match")
+                exit(1);
+            }
+            
+            let secretsData;
+            try {
+                const jsonSerializingCharacter = argv['secrets-ci'].slice(1,2);
+                const secretsUnserialized = argv['secrets-ci'].split(`${jsonSerializingCharacter} "`).join('"').split(`"${jsonSerializingCharacter}}`).join('"}');
+                secretsData = JSON.parse(secretsUnserialized)
```

Comment:
Is this to deal with the `n` character seen in the action logs?

Reply:
related to double encoding which is now removed